### PR TITLE
Update DeepSpeed Training example launcher

### PR DESCRIPTION
**Pull request by cassieesvelt**
# Description
Use pytorch instead of deepspeed launcher for deepspeed training example. This avoids having to set up ssh.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/jobs/deepspeed/deepspeed-training/job.yml [hunk 1]
```diff
@@ -1 +1,29 @@
...
+experiment_name: DistributedJob-DeepsSpeed-Training-cifar
+display_name: deepspeed-training-example
+code: src
+environment:
+  build:
+    path: docker-context
+environment_variables:
+  AZUREML_COMPUTE_USE_COMMON_RUNTIME: 'True'
+  AZUREML_COMMON_RUNTIME_USE_INTERACTIVE_CAPABILITY: 'True'
+  AZUREML_SSH_KEY: 'generated-key'
```
Comment (rtanase) [inline cli/jobs/deepspeed/deepspeed-training/job.yml:16]: AZUREML_SSH_KEY seems redundant now, right?

File: cli/jobs/deepspeed/deepspeed-training/job.yml [hunk 2]
```diff
@@ -1 +1,29 @@
...
+command: python train.py --with_aml_log=True --deepspeed --deepspeed_config ds_config.json
+
+experiment_name: DistributedJob-DeepsSpeed-Training-cifar
+display_name: deepspeed-training-example
+code: src
+environment:
+  build:
+    path: docker-context
+environment_variables:
+  AZUREML_COMPUTE_USE_COMMON_RUNTIME: 'True'
```
Comment (rtanase) [inline cli/jobs/deepspeed/deepspeed-training/job.yml:14]: can you pls confirm with @srsaggam whether the other 2 env variables are still needed?

Comment (srsaggam) [reply cli/jobs/deepspeed/deepspeed-training/job.yml:14]: We shouldn't be needing these variables anymore.

File: cli/jobs/deepspeed/deepspeed-training/job.yml (hunk 1 above)
Comment (cassieesvelt) [reply cli/jobs/deepspeed/deepspeed-training/job.yml]: yeah, I removed it.

**Review by savitamittal1 — APPROVED** [review]

**Review by rtanase — APPROVED** [review]

**Pull request merged**